io/include/pcl/io/file_io.h

```diff
@@ -324,90 +324,107 @@ namespace pcl
+           reinterpret_cast<char*>(&value),
+           sizeof(std::int8_t));
   }
 
-  template <> inline void
-  copyStringValue<std::uint8_t> (const std::string &st, pcl::PCLPointCloud2 &cloud,
-                           pcl::index_t point_index, unsigned int field_idx, unsigned int fields_count)
+  template <>
+  inline void
+  copyStringValue<std::uint8_t>(const std::string& st,
+                                pcl::PCLPointCloud2& cloud,
+                                pcl::index_t point_index,
+                                unsigned int field_idx,
+                                unsigned int fields_count,
+                                std::istringstream& is)
   {
     std::uint8_t value;
-    if (boost::iequals(st, "nan"))
-    {
-      value = static_cast<std::uint8_t> (std::numeric_limits<int>::quiet_NaN ());
-      cloud.is_dense = false;
-    }
-    else
-    {
-      int val;
-      std::istringstream is (st);
-      is.imbue (std::locale::classic ());
-      //is >> val;  -- unfortunately this fails on older GCC versions and CLANG on MacOS
-      if (!(is >> val))
-        val = static_cast<int> (atof (st.c_str ()));
-      value = static_cast<std::uint8_t> (val);
+    int val;
+    is.str(st);
+    // is >> val;  -- unfortunately this fails on older GCC versions and CLANG on
+    // MacOS
+    if (!(is >> val)) {
+      val = static_cast<int>(atof(st.c_str()));
     }
+    value = static_cast<std::uint8_t>(val);
 
-    memcpy (&cloud.data[point_index * cloud.point_step + 
-                        cloud.fields[field_idx].offset + 
-                        fields_count * sizeof (std::uint8_t)], reinterpret_cast<char*> (&value), sizeof (std::uint8_t));
+    memcpy(&cloud.data[point_index * cloud.point_step + cloud.fields[field_idx].offset +
+                       fields_count * sizeof(std::uint8_t)],
+           reinterpret_cast<char*>(&value),
+           sizeof(std::uint8_t));
   }
+  } // namespace detail
 
+  /** \brief Copy one single value of type T (uchar, char, uint, int, float, double, ...) from a string
+    * 
+    * Uses aoti/atof to do the conversion.
+    * Checks if the st is "nan" and converts it accordingly.
+    *
+    * \param[in] st the string containing the value to convert and copy
+    * \param[out] cloud the cloud to copy it to
+    * \param[in] point_index the index of the point
+    * \param[in] field_idx the index of the dimension/field
+    * \param[in] fields_count the current fields count
+    */
+  template <typename Type> inline void
+  copyStringValue (const std::string &st, pcl::PCLPointCloud2 &cloud,
+                   pcl::index_t point_index, unsigned int field_idx, unsigned int fields_count)
+  {
+    std::istringstream is;
+    is.imbue (std::locale::classic ());
+    detail::copyStringValue<Type> (st, cloud,point_index, field_idx, fields_count, is);
+  }
+  template <typename Type> inline void
+  copyStringValue (const std::string &st, pcl::PCLPointCloud2 &cloud,
```

Comment:
This new functions lacks the documentation, especially regarding `is`. Also: "Uses aoti/atof to do the conversion." isn't really correct anymore.